ext/storage_sources/azure_store/azure_connection.cpp

```diff
@@ -170,21 +170,25 @@ azure_connection::object_exists(
     exists = false;
     std::string obj = _bucket_prefix + object_key;
     object_size = 0;
+    Azure::Storage::Blobs::ListBlobsOptions blob_parameters;
 
-    auto list_blob_response = _azure_client.ListBlobs();
-
-    for (const auto blob_item : list_blob_response.Blobs) {
-        // Check if object exists.
-        if (blob_item.Name.compare(obj) == 0) {
-            // Check if object is deleted and has not been cleared by garbage collection.
-            if (blob_item.IsDeleted) {
-                return -1;
+    blob_parameters.Prefix = _bucket_prefix;
```

Comment:
Is there a reason we need to loop through all blobs starting with `_bucket_prefix` and can't use `_bucket_prefix + object_key` instead?

Reply:
good point! [c7dc520](https://github.com/wiredtiger/wiredtiger/pull/8984/commits/c7dc52029a313d3bbf1f7a35021935137612caa3)